### PR TITLE
Add chart resource update methods

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/1902

Added helm service stub method for update, will implement it next.

### CODE REVIEW

File: service/chartconfig/v1/resource/chart/update.go [hunk 1]
```diff
@@ -2,14 +2,81 @@ package chart
...
+
+	if currentChartState.ReleaseVersion != "" && desiredChartState.ReleaseVersion != currentChartState.ReleaseVersion {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "the chart has to be updated")
+
+		return &desiredChartState, nil
+	} else {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "the chart does not have to be updated")
+	}
+
+	return &ChartState{}, nil
```
Comment (xh3b4sd) [inline service/chartconfig/v1/resource/chart/update.go]: Is there a special reason why we want to return an initialized pointer here without any information? I think in such cases we should return nil. The handling of nil is way easier to check against. Getting initialized pointers without any values should be an error. The resource wants to try to do something useful and when it gets something which is actually nothing it cannot do anything. Getting nothing (`nil`) means to do nothing, which is straight forward.

File: service/chartconfig/v1/appr/appr.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func New(config Config) (*Client, error) {
 }
 
 // GetReleaseVersion queries CNR for the release version of the chart
-// represented by the given custom object (including channel info).
+// represented by the given name and channel.
```
Comment (rossf7) [inline service/chartconfig/v1/appr/appr.go]: <3

File: service/chartconfig/v1/resource/chart/update.go (hunk 1 above)
Comment (rossf7) [reply service/chartconfig/v1/resource/chart/update.go]: +1 I think returning nil is cleaner here.

File: service/chartconfig/v1/resource/chart/update.go [hunk 3]
```diff
@@ -2,14 +2,81 @@ package chart
...
+	name := key.ChartName(customObject)
+	releaseName := chartState.ReleaseName
+	channel := chartState.ChannelName
+
+	tarballPath, err := r.apprClient.PullChartTarball(name, channel)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.helmClient.UpdateReleaseFromTarball(releaseName, tarballPath)
```
Comment (kopiczko) [inline service/chartconfig/v1/resource/chart/update.go]: What if newUpdateChange returns empty ChartState?

File: service/chartconfig/v1/resource/chart/update.go (hunk 1 above)
Comment (fgimenez) [reply service/chartconfig/v1/resource/chart/update.go]: Agree, done PTAL.

File: service/chartconfig/v1/resource/chart/update.go (hunk 3 above)
Comment (rossf7) [reply service/chartconfig/v1/resource/chart/update.go]: Agreed if the ChartState is empty no action should be taken.

Comment (fgimenez) [reply service/chartconfig/v1/resource/chart/update.go]: Done, currently `toChartState` returns an empty `ChartState` on nil input, i've changed `ApplyUpdateChange` to check for an empty `ReleaseName` field, PTAL.

Comment (rossf7) [reply service/chartconfig/v1/resource/chart/update.go]: Thanks LGTM.

**Pull request merged**